go/vt/vtgate/engine/route.go

```diff
@@ -644,6 +715,27 @@ func (route *Route) processGenerate(vcursor VCursor, bindVars map[string]*queryp
 	return insertID, nil
 }
 
+// insertUpdatedVindexEntries inserts vindex entries are being changed by an update statement
```

Comment:
This is a really good idea. I will merge them. It was remanent of when we were focused on trying to make `insert index -> update value -> delete index`. Now that we decided that it doesn't really matter this can be merged. 

After we changed the order of the statements to be `delete index -> insert index -> update value`, the last problem is not an issue any more. I can also add the optimization so we don't even have to insert/delete to the index lookup table when is not changing.